### PR TITLE
[#478] feat: reset button

### LINKED ISSUE #478: Make reset button reset editor to initial state with a confirmation dialog

### PR DESCRIPTION
**Pull request by Tylopilus**
adds functionality to the reset button.

Closes #478

### CODE REVIEW

File: packages/monaco/src/EditorHooks.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { useEffect } from 'react';
+
+export function useResetEditor(): { add: (cb: () => void) => void; dispatch: () => void } {
+  const ev = new CustomEvent('resetCode');
+
+  function add(cb: () => void) {
```
Comment (bautistaaa) [inline packages/monaco/src/EditorHooks.ts:6]: i think a more appropriate name is subscribe.

out of scope but could probably make this more generic by being able to pass event name

File: packages/monaco/src/EditorHooks.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { useEffect } from 'react';
```
Comment (bautistaaa) [inline packages/monaco/src/EditorHooks.ts:1]: we kebab case our file names but i'd argue you can just put this in the index file unless you see this growing

Comment (bautistaaa) [reply packages/monaco/src/EditorHooks.ts:1]: oh i see its a separate entry point for importing

File: packages/monaco/src/EditorHooks.ts (hunk 1 above)
Comment (Tylopilus) [reply packages/monaco/src/EditorHooks.ts:6]: I agree with naming it subscribe but I question if we even should follow this or use zustand and create a store as we already have it as a dep. 

Depending on the other editor functionalities. But as these are just signals _I think_ a simple solution here is sufficient.

Comment (Tylopilus) [reply packages/monaco/src/EditorHooks.ts]: I changed it to be more generic. I don't know if this is common to setup like this or if hooks being factories is more common.

**Review by PickleNik — APPROVED** [review]
lgtm

**Pull request merged**